### PR TITLE
controller: Add a `testrender` command

### PR DESCRIPTION
**Pull request by cgwalters**
This will be used for unit testing; it accepts a `ct/` directory of
`.yaml` CT files (a bit like our builtin templates but without
the actual templating), as well as an `inline/` subdirectory,
and generates a MachineConfig as JSON.

The value of `inline/` is that for world-readable regular files
you don't need to indirect through YAML.

### CODE REVIEW

**Comment by cgwalters** [comment]
```
$ _output/linux/amd64/machine-config-controller testrender --srcdir ./test/daemon/mc/ | jq
{
  "metadata": {
    "name": "testmc",
    "creationTimestamp": null,
    "labels": {
      "machineconfiguration.openshift.io/role": "worker"
    }
  },
  "spec": {
    "osImageURL": "",
    "config": {
      "ignition": {
        "config": {},
        "security": {
          "tls": {}
        },
        "timeouts": {},
        "version": "2.2.0"
      },
      "networkd": {},
      "passwd": {},
      "storage": {
        "files": [
          {
            "filesystem": "root",
            "path": "/etc/tmpfiles.d/cleanup-cni.conf",
            "contents": {
              "source": "data:,r%20%2Fetc%2Fkubernetes%2Fcni%2Fnet.d%2F80-openshift-network.conf%0Ar%20%2Fetc%2Fkubernetes%2Fcni%2Fnet.d%2F10-ovn-kubernetes.conf%0Ar%20%2Fetc%2Fkubernetes%2Fcni%2Fnet.d%2F00-multus.conf%0A",
              "verification": {}
            },
            "mode": 420
          },
          {
            "filesystem": "root",
            "path": "/etc/sysctl.d/forward.conf",
            "contents": {
              "source": "data:,net.ipv4.ip_forward%20%3D%201%0A",
              "verification": {}
            },
            "mode": 420
          },
          {
            "filesystem": "root",
            "path": "/usr/share/foo.txt",
            "contents": {
              "source": "data:,foo%0A",
              "verification": {}
            },
            "mode": 420
          }
        ]
      },
      "systemd": {}
    }
  }
}

```

**Comment by cgwalters** [comment]
Basically I want to add some shell-script based tests that run the MCD in a once-from like mode and just test `updateFiles`.

Specifically with an eye to fixing https://github.com/openshift/machine-config-operator/issues/782
(which would also require stubbing out a more ostree-like system with `usr/etc` etc.)

File: cmd/machine-config-controller/testrender.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
+package main
```
Comment (runcom) [inline cmd/machine-config-controller/testrender.go]: it's ok since we're not compiling this anywhere but could we move this out of cmd :thinking:

**Comment by kikisdeliveryservice** [comment]
In an effort to clean up the MCO repo, closing old open PRs with no recent activity.  

Feel free to reopen.

**Pull request closed without merging**